fairroot/base/steer/FairRunAna.cxx

```diff
@@ -200,20 +203,18 @@ void FairRunAna::Init()
 
     // Assure that basic info is there for the run
     //  if(par && fInputFile) {
+    auto* evtHeader = GetEventHeader();
     if (par && fInFileIsOpen) {
 
         LOG(info) << "Parameter and input file are available, Assure that basic info is there for the run!";
         fRootManager->SpecifyRunId();
 
-        //    fEvtHeader = GetEventHeader();
-        GetEventHeader();
-
-        FillEventHeader();
+        FillEventHeader(evtHeader);
 
-        fRunId = GetEvtHeaderRunId();
+        fRunId = GetEvtHeaderRunId(evtHeader);
 
         // Copy the Event Header Info to Output
-        fEvtHeader->Register(GetSink() ? fStoreEventHeader : false);
+        evtHeader->Register(fStoreEventHeader);
```

Comment:
We added the `GetSink` checks on purpose. Can you please keep them?

Reply:
Sure